### PR TITLE
165967138 Improve Test On Profile Feature

### PR DESCRIPTION
**Pull request by chikeozulumba**
#### Description
The tests folders are to properly structured.
Test the state, reducers and actions properly.
#### Type of change
- [ ] Bug Fix (a non-breaking change which fixes an issue)
- [ ] New Feature (a non-breaking change that adds functionality)
- [x] New Chore (a non-breaking change that adds functionality)
- [ ] Breaking Change (a fix or feature which would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update
#### How Has This Been Tested?
- [ ] End to End
- [x] Integration
#### Checklist:
- [x] Unit test
- [x] Code coverage
#### PT
[#165967138](https://www.pivotaltracker.com/story/show/165967138)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -74,4 +74,7 @@ typings/
 
 # Build folder
 dist
+
 .DS_Store
+
+__snapshots__
```
Comment (micah-akpan) [inline .gitignore]: We should have a new line here

File: package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
   "author": "",
   "license": "ISC",
   "scripts": {
-    "test": "jest | coveralls",
+    "test": "jest",
```
Comment (micah-akpan) [inline package.json:30]: I think I understand why this change was made. But can you explain here, so we can understand why you are not piping the test results to coveralls in the `test script` anymore

File: package.json [hunk 3]
```diff
@@ -109,9 +112,7 @@
     "collectCoverage": true,
     "coverageDirectory": "coverage",
     "coverageReporters": [
-      "lcov",
-      "text",
-      "text-lcov"
+      "text"
```
Comment (micah-akpan) [inline package.json]: @chikeozulumba  well done on this fix

File: src/components/__tests__/activity.component.test.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+afterEach(cleanup);
+
+describe('Should render the component', () => {
+  test('<Activity /> component', () => {
+    const activity = render(<Activity />);
+    expect(activity).toBeTruthy();
+    expect(activity).toMatchSnapshot();
+  });
+});
```
Comment (micah-akpan) [inline src/components/__tests__/activity.component.test.js:13]: we should have a new line here too

File: src/components/__tests__/bookmark.component.test.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+afterEach(cleanup);
+
+describe('Should render the component', () => {
+  test('<Bookmark /> component', () => {
+    const bookmark = render(<Bookmark />);
+    expect(bookmark).toBeTruthy();
+    expect(bookmark).toMatchSnapshot();
+  });
+});
```
Comment (micah-akpan) [inline src/components/__tests__/bookmark.component.test.js:13]: @chikeozulumba we should have a new line here too

File: src/components/__tests__/like.component.test.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+afterEach(cleanup);
+
+describe('Should render the component', () => {
+  test('<Like /> component', () => {
+    const like = render(<Like />);
+    expect(like).toBeTruthy();
+    expect(like).toMatchSnapshot();
+  });
+});
```
Comment (micah-akpan) [inline src/components/__tests__/like.component.test.js:13]: we also need a new line here

File: src/components/__tests__/loader.component.test.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+afterEach(cleanup);
+
+describe('Should render the component', () => {
+  test('<ContentLoader /> component', () => {
+    const loader = render(<ContentLoader />);
+    expect(loader).toBeTruthy();
+    expect(loader).toMatchSnapshot();
+  });
+});
```
Comment (micah-akpan) [inline src/components/__tests__/loader.component.test.js:16]: we need a new line here

File: src/components/__tests__/nav.component.test.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+afterEach(cleanup);
+
+describe('Should render the component', () => {
+  test('<Navigation /> component', () => {
+    const navigation = render(<Router><Navigation /></Router>);
+    const { debug, getByText } = navigation;
+    expect(navigation).toBeTruthy();
+    expect(navigation.container.firstChild).toMatchSnapshot();
+  });
+});
```
Comment (micah-akpan) [inline src/components/__tests__/nav.component.test.js:17]: we need a new line here too

File: src/pages/__tests__/Profile.page.test.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
+
+import React from 'react';
+import { shallow } from 'enzyme';
```
Comment (micah-akpan) [inline src/pages/__tests__/Profile.page.test.js:3]: I thought we were asked not to use `enzyme`, since it doesn't work well with React hooks?

File: package.json [hunk 10]
```diff
@@ -53,6 +53,8 @@
     "babel-preset-env": "^1.7.0",
     "babel-preset-react": "^6.24.1",
     "coveralls": "^3.0.3",
+    "enzyme": "^3.9.0",
```
Comment (Raymond-Osy) [inline package.json:57]: I thought we are required to use react-testing-library for testing @chikeozulumba , you might want to revisit this.

Comment (chikeozulumba) [reply package.json:57]: @Raymond-Osy 
We are not required to use a particular library.

**Review by micah-akpan — DISMISSED** [review]

File: package.json (hunk 2 above)
Comment (chikeozulumba) [reply package.json:30]: @micah-akpan That was an over sight

**Review by Temmyogunbo — APPROVED** [review]
